analyzer/analyzer.go

```diff
@@ -54,6 +54,48 @@ func newPerfSprint() *perfSprint {
 	}
 }
 
+// checker is a type of check that can be performed by the analyzer.
+// A checker can activate or deactivate specific rules.
+type checker = string
+
+const (
+	// checkerErrorFormat checks for error formatting.
+	checkerErrorFormat checker = "error-format"
+	// checkerIntegerFormat checks for integer formatting.
+	checkerIntegerFormat checker = "integer-format"
+	// checkerStringFormat checks for string formatting.
+	checkerStringFormat checker = "string-format"
+	// checkerBoolFormat checks for bool formatting.
+	checkerBoolFormat checker = "bool-format"
+	// checkerHexFormat checks for hexadecimal formatting.
+	checkerHexFormat checker = "hex-format"
+)
+
+// checkerOption is a type a specific [checker] that can be activated or deactivated for a [checker].
+type checkerOption = checker
+
+const (
+	// checkerOptionIntConversion optimizes even if it requires an int or uint type cast
+	// This option is only available when [CheckerIntegerFormat] is activated.
+	checkerOptionIntConversion checkerOption = "int-conversion"
+	// checkerOptionErrError optimizes into err.Error() even if it is only equivalent for non-nil errors
+	// This option is only available when [CheckerErrorFormat] is activated.
+	checkerOptionErrError checkerOption = "err-error"
+	// checkerOptionErrorf optimizes fmt.Errorf
+	// This option is only available when [CheckerErrorFormat] is activated.
+	checkerOptionErrorf checkerOption = "errorf"
+	// checkerOptionSprintf1 optimizes fmt.Sprintf with only one argument
+	// This option is only available when [CheckerStringFormat] is activated.
+	checkerOptionSprintf1 checkerOption = "sprintf1"
+	// checkerOptionStrConcat optimizes string concatenation
+	// This option is only available when [CheckerStringFormat] is activated.
+	checkerOptionStrConcat checkerOption = "strconcat"
```

Comment:
This looks unnecessary to define a string to use it only once...

Reply:
I iterated, I was using them at some point. In the diagnostic, but then I left them like this.

But I agree, now the logic of type and even constants for the suboption seems overkill.